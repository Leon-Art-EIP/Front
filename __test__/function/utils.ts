import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
