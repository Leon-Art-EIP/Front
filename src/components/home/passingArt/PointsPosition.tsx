import { Circle, CircleTwoTone } from "@mui/icons-material";

export interface IPointPositionsProps {
  nbrPoints: number;
  position: number;
}

export default function PointsPosition(props: IPointPositionsProps): JSX.Element {
  if (props.position > props.nbrPoints) {
    return <></>;
  }
  return (
    <div className="flex gap-1">
      {Array.from(Array(props.nbrPoints).keys()).map((_, index) => {
        if (index + 1 === props.position) {
          return <CircleTwoTone key={`point-${index}`} className="text-primary" style={{ fontSize: 12 }} />;
        }
        return <Circle key={`point-${index}`} className="text-secondary-disabled" style={{ fontSize: 12 }} />;
      })}
    </div>
  );
}
