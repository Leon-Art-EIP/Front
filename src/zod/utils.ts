export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
