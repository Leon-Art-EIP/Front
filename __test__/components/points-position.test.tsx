import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import PointsPosition from "../../src/components/home/passingArt/PointsPosition";

test("PointsPosition 0 point snapshot", () => {
  testSnapshot(<PointsPosition nbrPoints={0} position={0} />);
});

test("PointsPosition snapshot", () => {
  testSnapshot(<PointsPosition nbrPoints={10} position={5} />);
});
