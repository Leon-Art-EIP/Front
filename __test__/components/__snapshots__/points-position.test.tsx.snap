// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PointsPosition 0 point snapshot 1`] = `
<div
  className="flex gap-1"
/>
`;

exports[`PointsPosition snapshot 1`] = `
<div
  className="flex gap-1"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-secondary-disabled css-i4bv87-MuiSvgIcon-root"
    data-testid="CircleIcon"
    focusable="false"
    style={
      {
        "fontSize": 12,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-secondary-disabled css-i4bv87-MuiSvgIcon-root"
    data-testid="CircleIcon"
    focusable="false"
    style={
      {
        "fontSize": 12,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-secondary-disabled css-i4bv87-MuiSvgIcon-root"
    data-testid="CircleIcon"
    focusable="false"
    style={
      {
        "fontSize": 12,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-secondary-disabled css-i4bv87-MuiSvgIcon-root"
    data-testid="CircleIcon"
    focusable="false"
    style={
      {
        "fontSize": 12,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-primary css-i4bv87-MuiSvgIcon-root"
    data-testid="CircleTwoToneIcon"
    focusable="false"
    style={
      {
        "fontSize": 12,
      }
    }
    viewBox="0 0 24 24"
  >
    <circle
      cx="12"
      cy="12"
      opacity=".3"
      r="8"
    />
    <path
      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-secondary-disabled css-i4bv87-MuiSvgIcon-root"
    data-testid="CircleIcon"
    focusable="false"
    style={
      {
        "fontSize": 12,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-secondary-disabled css-i4bv87-MuiSvgIcon-root"
    data-testid="CircleIcon"
    focusable="false"
    style={
      {
        "fontSize": 12,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-secondary-disabled css-i4bv87-MuiSvgIcon-root"
    data-testid="CircleIcon"
    focusable="false"
    style={
      {
        "fontSize": 12,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-secondary-disabled css-i4bv87-MuiSvgIcon-root"
    data-testid="CircleIcon"
    focusable="false"
    style={
      {
        "fontSize": 12,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-secondary-disabled css-i4bv87-MuiSvgIcon-root"
    data-testid="CircleIcon"
    focusable="false"
    style={
      {
        "fontSize": 12,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
    />
  </svg>
</div>
`;
